### PR TITLE
IGAPP-629: Type sagas

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/sagas/__tests__/watchFetchCategory.spec.ts [hunk 1]
```diff
@@ -159,7 +159,7 @@ describe('watchFetchCategories', () => {
             categoriesMap: categories,
             resourceCache: resources,
             path: initialPath,
-            cityLanguages: languages,
+            cityLanguages: [],
```
Comment (klinzo) [inline native/src/sagas/__tests__/watchFetchCategory.spec.ts:162]: Why these changes?

File: native/src/sagas/fetchResourceCache.ts [hunk 2]
```diff
@@ -27,21 +22,21 @@ const createErrorMessage = (fetchResult: FetchResultType) => {
   )
 }
 
-function* watchOnProgress(): Generator<StrictEffect, void, number> {
+function* watchOnProgress() {
```
Comment (klinzo) [inline native/src/sagas/fetchResourceCache.ts:25]: Types for void are omitted?

File: native/src/sagas/loadCityContent.ts [hunk 3]
```diff
@@ -169,15 +161,15 @@ export default function* loadCityContent(
...
+      const languageAvailable = yield* call(prepareLanguages, dataContainer, newCity, newLanguage, shouldUpdate)
 
       if (!languageAvailable) {
         return false
       }
     }
 
     const { featureFlags } = buildConfig()
-    const [categoriesMap, events, _unusedPois] = (yield all([
+    const [categoriesMap, events, _unusedPois] = (yield* all<any>([
```
Comment (klinzo) [inline native/src/sagas/loadCityContent.ts:172]: Do we have to use any here?

File: native/src/sagas/watchContentLanguageSwitch.ts [hunk 4]
```diff
@@ -34,31 +36,29 @@ export function* switchContentLanguage(
         false
       )
     )
-    const [categories, resourceCache, events, pois] = (yield all([
+    const [categories, resourceCache, events, pois] = (yield* all<any>([
```
Comment (klinzo) [inline native/src/sagas/watchContentLanguageSwitch.ts:39]: Do we have to use any here?

File: native/src/sagas/watchFetchCategory.ts [hunk 5]
```diff
@@ -21,45 +18,28 @@ import { LanguageResourceCacheStateType } from '../redux/StateType'
  * @param routeCity The key of the current route
  * @returns true if the fetch corresponds to a peek
  */
-function* isPeeking(routeCity: string): Generator<SelectEffect, void> {
-  yield select(state =>
+function* isPeeking(routeCity: string): SagaGenerator<boolean> {
+  return yield* select(state =>
```
Comment (klinzo) [inline native/src/sagas/watchFetchCategory.ts:22]: :thinking:

**Review by klinzo — APPROVED** [review]
Very very nice, well done!

Not tested

**Review by SteffiStuffel — APPROVED** [review]

File: native/src/sagas/watchFetchCategory.ts (hunk 5 above)
Comment (Taggotty) [reply native/src/sagas/watchFetchCategory.ts:22]: before we actually did call isPeeking, but it would always return `undefined`, of course, as return was missing. That's also what you see in the tests. If a route isPeeking the languages are not fetched for example.

Comment (klinzo) [reply native/src/sagas/watchFetchCategory.ts:22]: Yes, was just surprised that we did not return it, so the logic was faulty all the time :see_no_evil:

File: native/src/sagas/fetchResourceCache.ts (hunk 2 above)
Comment (Taggotty) [reply native/src/sagas/fetchResourceCache.ts:25]: no. types for non-exported functions can be omitted.

**Comment by klinzo** [comment]
@Taggotty I like the .all with the object syntax. I think there were one or two more occurrences of this `any` array syntax, could you check please?
